docs/preparing_recipes_for_conan2.md

```diff
@@ -0,0 +1,373 @@
+> not affect current CMake 1.X generators (``cmake``, ``cmake_multi``, ``cmake_find_package`` and
+> ``cmake_find_package_multi``) at all.
+
+### CMakeDeps
+
+### Update required_conan_version to ">=1.43.0"
+
+If you set the property ``cmake_target_name`` in the recipe, the Conan minimum
+required version should be updated to 1.43. 
+
+```python
+
+required_conan_version = ">=1.43.0"
+
+class GdalConan(ConanFile):
+    name = "gdal"
+    ...
+```
+
+The reason for this change is that in Conan versions previous to 1.43 the
+``cmake_target_name`` values were not the final CMake target names. Those values were
+completed by Conan, adding namespaces automatically the final target names. After 1.43
+``cmake_target_name`` sets the **complete target name** that is added to the ``.cmake``
+files generated by Conan. Let's see an example:
+
+```python
+class GdalConan(ConanFile):
+    name = "gdal"
+    ...
+    def package_info(self):
+        # Before 1.43 -> Conan adds GDAL:: namespace -> Creates target with name GDAL::GDAL
+        # self.cpp_info.set_property("cmake_target_name", "GDAL")
+
+        # After 1.43 -> Conan creates target with name GDAL::GDAL
+        self.cpp_info.set_property("cmake_target_name", "GDAL::GDAL")
+```
+
+### Translating .names information to cmake_target_name, cmake_module_target_name and cmake_file_name
+
+To translate the ``.names`` information to the new model there are some important things to
+take into account:
+
+* The value of the ``.names`` attribute value in recipes is just a part of the final
+  target name for CMake generators. Conan will complete the rest of the target name by
+  pre-pending a namespace (with ``::`` separator) to the ``.names`` value. This namespace takes
+  the same value as the ``.names`` value. Let's see an example: 
+
+```python
+class SomePkgConan(ConanFile):
+    name = "somepkg"
+    ...
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "some-pkg"
+        self.cpp_info.names["cmake_find_package_multi"] = "some-pkg"
+        ...
+```
+
+This recipe generates the target ``some-pkg::some-pkg`` for both the
+``cmake_find_package`` and the ``cmake_find_package_multi`` generators. Also, please
+remember that if no ``.names`` attribute were set, Conan would create the target
+``somepkg::somepkg`` for both generators by default.
+
+As we explained before, the ``cmake_target_name`` sets the **complete target name**, so, 
+to translate this information to the new model we should add the following lines:
+
+```python
+class SomePkgConan(ConanFile):
+    name = "somepkg"
+    ...
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "some-pkg"
+        self.cpp_info.names["cmake_find_package_multi"] = "some-pkg"
+        # CMakeDeps does NOT add any namespace automatically
+        self.cpp_info.set_property("cmake_target_name", "some-pkg::some-pkg") 
```

Comment:
As the name of the package is `somepkg`, shouldn't the namespace be... ?
```suggestion
        self.cpp_info.set_property("cmake_target_name", "somepkg::some-pkg") 
```

Reply:
The thing is that if you set the .names attribute to something, Conan uses that value for both the namespace and the part after it. So no matter the name of the package, if you set `self.cpp_info.names["cmake_find_package"] = "foo"` it generates the target `foo::foo`

Reply:
oh, ok. Maybe it was just me don't remembering how `.names` worked. I think it is fine. Thanks for the clarification